### PR TITLE
electron_24: build from source

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes
I tested this by building bitwarden desktop and successful launching and using it.
This removes the last electron-bin from my system. I first didn't thought it was that easy but it was so I am upstreaming it. :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/electron/info.json [hunk 1]
```diff
@@ -2554,5 +2554,826 @@
...
+                    "version": "2023-02-17",
+                    "url": "https://gn.googlesource.com/gn",
+                    "rev": "b25a2f8c2d33f02082f0f258350f5e22c0973108",
+                    "sha256": "075p4jwk1apvwmqmvhwfw5f669ci7nxwjq9mz5aa2g5lz4fkdm4c"
+                }
+            }
+        },
+        "chromium_npm_hash": "sha256-e3CEKcUEFXDHeJ6Ol7qaabIfSbyRYobi0+nQmUbfBy4=",
+        "electron_yarn_hash": "147zj0k97xx5gjqqw80hd1fvm87wavps5p3zh0q72vcafaayc0hf"
+>>>>>>> 05aed5329365 (electron_24: build from source)
```
Comment (yu-re-ka) [inline pkgs/development/tools/electron/info.json:3377]: this includes a merge oopsie

**Comment by yu-re-ka** [comment]
Electron 24 goes EOL in 6 days

**Comment by SuperSandro2000** [comment]
> Electron 24 goes EOL in 6 days.

Sadly those EOL cycles are not followed by downstream applications like bitwarden https://github.com/bitwarden/clients/blob/master/package.json#L91 . I tried creating a feature request to maybe draw some attention to this https://community.bitwarden.com/t/track-electron-releases-more-closely/58793

> I assume we want to remove it once Electron 27 has been released

Not really and just updating it under an applications toes just because it recently got EOL is also not a good option.

**Comment by yu-re-ka** [comment]
I'm not going to decide this. @vcunat @primeos can have a look if we want to have 4 supported electron versions (= 8 chromium versions behind chromium stable) packaged at once.

**Comment by vcunat** [comment]
OfBorg says that eval fails.  When 24 goes out of support, I expect it to soon be marked with `knownVulnerabilities`.  From that point it won't be getting new binaries on cache.nixos.org.  For those that still want to use it, binary electron will be easier in that phase.

**Pull request closed without merging**